lightning/src/ln/channelmanager.rs

```diff
@@ -5311,6 +5344,32 @@ where
 				msg.temporary_channel_id.clone()));
 		}
 
+		let channel_id = msg.temporary_channel_id;
+		let channel_exists = peer_state.has_channel(&channel_id);
+		if channel_exists {
+			return Err(MsgHandleErrInternal::send_err_msg_no_close("temporary_channel_id collision for the same peer!".to_owned(), msg.temporary_channel_id.clone()));
+		}
+
+		// If we're doing manual acceptance checks on the channel, then defer creation until we're sure we want to accept.
+		if self.default_configuration.manually_accept_inbound_channels {
+			let mut pending_events = self.pending_events.lock().unwrap();
+			pending_events.push_back((events::Event::OpenChannelRequest {
+				temporary_channel_id: msg.temporary_channel_id.clone(),
+				counterparty_node_id: counterparty_node_id.clone(),
+				funding_satoshis: msg.funding_satoshis,
+				push_msat: msg.push_msat,
+				channel_type: msg.channel_type.clone().unwrap(),
+			}, None));
+			peer_state.inbound_channel_request_by_id.insert(channel_id, InboundChannelRequest {
+				counterparty_node_id: *counterparty_node_id,
+				open_channel_msg: msg.clone(),
+			});
+			return Ok(());
+		}
+
+		// Otherwise create the channel right now.
+		let user_channel_id = u128::from_be_bytes(random_bytes);
+		let outbound_scid_alias = self.create_and_insert_outbound_scid_alias();
```

Comment:
Hmm. @wpaulino do the error conditions below leave a dangling entry in the table of outbound SCID aliases? Should we call `InboundV1Channel::new` here with a zero `outbound_scid_alias` and then assign it after we've cleared the checks below?